extensions/map-layers/src/mapLayers.ts

```diff
@@ -44,28 +44,26 @@ export class MapLayersUI {
 
     MapLayersUI._iTwinConfig = config?.iTwinConfig;
 
-    MapLayersUI._uiItemsProviders.push(
-      new MapLayersUiItemsProvider({ ...config?.mapLayerOptions })
-    );
+    const mlProvider = new MapLayersUiItemsProvider({ ...config?.mapLayerOptions });
+    const mlProviderId = config?.mapLayerProviderOverrides?.providerId ?? mlProvider.id;
+    MapLayersUI._uiItemsProvidersId.push(mlProviderId);
+    UiItemsManager.register(mlProvider, config?.mapLayerProviderOverrides);
```

Comment:
we need to take a stance on what best practice is on this soon. should ui items providers be registered independently of the init call? thats how i've done it in the vcr repo

Reply:
@mdastous-bentley may have more input of this and if it make sense to independently register the providers.

Reply:
For legacy reasons, we have always registered it part of the initialize.  I don't really mind if we stop doing it, especially if every other component does the same.

However, if we do, this will be a breaking change and  hosting applications will be forced to react to it.  Maybe we should considering making registering optional like we de with the FeatureInfo widget?  Any other suggestion to solve this?

Reply:
As Arun said, we really need best practices, so all widgets behave the same way. In any case, that should be addressed in separate PR.

Reply:
Bill and I talked this over before his vacation. I've broken out the items provider registration and added an optional prop in the interface to delay initialization. This should allow DR (or any app) the opportunity to fine-tune the overrides on the items providers, if they like. If they don't have those requirements, MapLayers and FeatureInfo will work as before.
Please review and let me know what you think.

Reply:
+ @kckst8 for his thoughts as the new extension dev lead

i like this change nancy, i always prefer to defer registering providers until theyre needed
re defaults, idk if they were ever right to begin with, just what went out in the initial api that we went with, i believe it's only DR who consumes this. we dont really have any documentation for this so we've been making breaking changes to this pkg (after alerting the DR team) periodically. so i wouldn't be opposed to changing/removing the default behavior

Reply:
Im ok with the change..  I guess we just need to decide if we break the default behavior or not... as Arun said, we already broke the API.

Reply:
I don't think we break the API if we leave the defaul tin place, do we? We added an optional prop to the config interface.
If we remove the default behavior and the prop, the API is broken, though.

Reply:
Michel meant we already broke the api in 3.1 😅

Reply:
yes exactly; I meant we previously broke the API

Reply:
But of course we did. :D. OK -- so leave the default or force a separate provider registration?

Reply:
I would say leave the default, we will have to the chance to revisit it in a next major release.  That time it will not require the API to be redesigned at least.